### DISCUSSION TITLE
Show diagnostics messages on hover

### DISCUSSION THREAD

**Discussion by ngaloppo** [LazyVim/LazyVim#550] [General]
I've turned off diagnostics virtual text by adjusting the lsp-config plugin options (as described [here](https://github.com/LazyVim/LazyVim/discussions/54#discussioncomment-4675766)). I'm looking for a solution to see the diagnostic text of the current line - this post seems to suggest that one can use a keybind to `vim.lsp.buf.hover()` like [here](https://github.com/samhh/dotfiles/blob/67f6bb5e4b3b293899273b8c3745778ad7029a94/headful/cfg/nvim/plugin/lsp.lua#L17). 

How would I achieve that in LazyVim? I see that currently `K` is assigned to a generic "Hover".

**Comment by Dolvur** [comment]
I think `<leader>cd` is bound to line diagnostics by default. You can see the default LSP keymaps LazyVim defines [here](https://github.com/LazyVim/LazyVim/blob/b984c0cff57177fd002f8927d9914ecdf562623c/lua/lazyvim/plugins/lsp/keymaps.lua)

**Reply by ngaloppo** [reply]
Thanks, I had no idea. That works for me!

**Comment by dvic** [comment]
How do you scroll inside that pop?

**Reply by folke** [reply]
`<c-f>` and `<c-b>`

**Reply by dvic** [reply]
I just checked, this does not work in my case.. Only if I press `<leader>cd` again, then the cursor moves inside the pop-up and the regular key bindings apply, in which case `<c-f>` and `<c-b>` of course do work. Is it supposed to work without pressing `<leader>cd` twice?

**Reply by dvic** [reply]
By the way: the same applies when I just use `K` to open `vim.lsp.buf.hover`

**Reply by folke** [reply]
Open a lua file, and add `vim.api.nvim_create_autocmd()`

Then the signature help or after pressing `K`, you can use `<c-f>` and `<c-b>` to scroll the doc window without focusing it.

This is a feature provided by Noice's lsp docs.

If it doesn't work for you, then some other plugin may be overriding those keymaps.

Check with `:verbose nmap <c-f>`

**Reply by dvic** [reply]
I tried this example with a clean `repro.lua` and indeed it works as you say. I boiled down the problem due to my rust setup (which I'm still fiddling with, see below). Do you happen to know this custom hover actions from rust-tools relate to the regular lsp hover?

In any case, thanks for your helping me figure this out👍!

```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        rust_analyzer = {},
      },
    },
    setup = {
      rust_analyzer = function(_, _)
        return true
      end,
    },
  },
  {
    "simrat39/rust-tools.nvim",
    dependencies = "neovim/nvim-lspconfig",
    config = function()
      local rt = require("rust-tools")

      rt.setup({
        server = {
          on_attach = require("lazyvim.util").on_attach(function(_, bufnr)
            -- Hover actions
            vim.keymap.set("n", "K, rt.hover_actions.hover_actions, { buffer = bufnr, desc = "Hover" })
            -- Code action groups
            vim.keymap.set(
              "n",
              "<Leader>ca",
              rt.code_action_group.code_action_group,
              { buffer = bufnr, desc = "Code Action" }
            )
          end),
        },
      })
    end,
  },
}
```
